### PR TITLE
Feat: Added user authentication and Cypress testing framework

### PR DESCRIPTION
**Pull request by bayfaub**
## Notes:

1. I opted to go with email and password sign up since "turning on the money funnel" may require more detailed user account information and will allow for easier account recovery in the future.
2. I haven't added unit-testing or e2e yet but that is coming with a future update to this PR.

## Features
1. Signup using Supabase Authentication(Email and Password) - Need to add tests
2. Login using Supabase Auth
3. Sign out
4. Added Cypress testing framework

### CODE REVIEW

File: app/root.tsx
Comment (Pinwheeler) [inline app/root.tsx]: with these whitespace changes, we should just get consistent on a spacing format. I like 2 spaces because it makes nested code easier to read. There's an argument to be made against making it too easy to nest code, but I still prefer the 2-space aesthetic.

I will leave it to you to either change your local or change the project spacing requirements

File: app/routes/home.tsx
Comment (Pinwheeler) [inline app/routes/home.tsx]: this edit, too, seems like it is more dramatic for having different spacing. We should look at the diff there before merging.

**Review by Pinwheeler — CHANGES_REQUESTED** [review]
Need to either
1. change local spacing formatting
2. change project spacing formatting and commit as separate commit before re-applying this diff.

File: app/api/signIn.ts
Comment (bayfaub) [inline app/api/signIn.ts]: @Pinwheeler What is your opinion on each argument having it's own line? Prettier does this when the argument exceeds the default printWidth. I'm curious if I should just make the printWidth larger or if this is fine.

File: app/routes/home.tsx
Comment (bayfaub) [reply app/routes/home.tsx]: I'm not sure if I addressed this comment with my spacing changes in the latest commit, let me know if not.

File: app/api/signIn.ts
Comment (Pinwheeler) [reply app/api/signIn.ts]: No real preference here. If you want to try it the other way, I'm game

File: app/routes/home.tsx
Comment (Pinwheeler) [inline app/routes/home.tsx]: the convention behind prefixing var names with `_` is to note them as private. Return statements should generally not include that. The preferred interface here is
```
return { action:_action, error}
```

File: app/util/supabase/supabase.server.ts
Comment (Pinwheeler) [inline app/util/supabase/supabase.server.ts]: best practice not to log cookies unless in a dev env

File: cypress/e2e/1-getting-started/todo.cy.js
Comment (Pinwheeler) [inline cypress/e2e/1-getting-started/todo.cy.js]: probably add these to the `.gitignore` if you want to keep them on the local filesystem

**Review by Pinwheeler — CHANGES_REQUESTED** [review]
Good looking code- a few minor changes and we should be g2g

File: cypress/e2e/1-getting-started/todo.cy.js
Comment (bayfaub) [reply cypress/e2e/1-getting-started/todo.cy.js]: Removed the examples from the project.

File: app/util/supabase/supabase.server.ts
Comment (bayfaub) [reply app/util/supabase/supabase.server.ts]: I went through and removed some other logs too.

File: app/routes/home.tsx
Comment (bayfaub) [reply app/routes/home.tsx]: Understood, thanks for the reminder.

**Review by Pinwheeler — APPROVED** [review]

**Pull request merged**
